### PR TITLE
Pl matching

### LINKED ISSUE #48: Support pl-matching

### PR DESCRIPTION
**Pull request by yangchristina**
Resolves #48

### CODE REVIEW

**Comment by firasm** [comment]
Hi @amurph24 - can I get you to first, review this PR for the [pl-matching element](https://prairielearn.readthedocs.io/en/latest/elements/#pl-matching-element), and then once that's done, update the test suite to add support for this element. Then, request a review from @Bluesy1.

----

@yangchristina - I've fixed the GH action failures you were seeing related to the problem_bank_helper versions and python versions. Sit tight for now while the PR is reviewed.

File: tests/test_question_templates/question_expected_outputs/instructor/q14_matching/q14_matching.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+---
+title: Distance travelled
+topic: Template
+author: Firas Moosvi
```
Comment (amurph24) [inline tests/test_question_templates/question_expected_outputs/instructor/q14_matching/q14_matching.md:4]:
```suggestion
author: Christina Yang
```

**Review by amurph24 — COMMENTED** [review]
It works as expected, and can the question can run locally on the docker shell. Code looks reasonable, Looks good to me!

**Comment by amurph24** [comment]
I looks like the test suite was already supportive of the `Matching` question type, the expected test output looked to be out of date with the current template.md file. Testing passes [here](https://github.com/open-resources/problem_bank_scripts/actions/runs/5432641406)

File: tests/test_question_templates/question_inputs/q15_matching/q15_matching.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
+---
+title: Distance travelled
+topic: Template
+author: Firas Moosvi
```
Comment (Bluesy1) [inline tests/test_question_templates/question_inputs/q15_matching/q15_matching.md:4]:
```suggestion
author: Christina Yang
```

File: tests/test_question_templates/question_expected_outputs/public/q15_matching/q15_matching.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
+---
+title: Distance travelled
+topic: Template
+author: Firas Moosvi
```
Comment (Bluesy1) [inline tests/test_question_templates/question_expected_outputs/public/q15_matching/q15_matching.md:4]:
```suggestion
author: Christina Yang
```

File: tests/test_question_templates/question_expected_outputs/instructor/q15_matching/q15_matching.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
+---
+title: Distance travelled
+topic: Template
+author: Firas Moosvi
```
Comment (Bluesy1) [inline tests/test_question_templates/question_expected_outputs/instructor/q15_matching/q15_matching.md:4]:
```suggestion
author: Christina Yang
```

**Review by Bluesy1 — COMMENTED** [review]
I accidentally clobbered @amurph24's suggestion, sorry - this is me readding those suggestions for him

I renamed the matching test case to q15 so we don't have two q14s - it looks good to me once you fix the author name, like was previously pointed out!

This is a good test case, and the implementation of the code seems solid for covering the addition, good job!

**Review by Bluesy1 — APPROVED** [review]
@firasm I'll throw this back to you now, it looks good across the board to me

**Review by firasm — APPROVED** [review]

**Comment by firasm** [comment]
Nice work everyone! Merging this in...

**Pull request merged**